src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -199,7 +199,30 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
 
     /// Helper method called by [`Self::read`]. Reads the current value as an int.
     fn read_int(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
-        todo!();
+        debug_assert!(self.encoded_value.ion_type() == IonType::Int);
+
+        let header = &self.encoded_value.header();
+        let representation = header.type_code();
+        let value = match (representation, header.length_code as usize) {
+            (OpcodeType::Integer, 0x0) => 0.into(),
+            (OpcodeType::Integer, n) => {
+                // We have n bytes following that make up our integer.
+                let (our_int, _) = self.input.consume(1).read_fixed_int(n)?;
+                our_int.value().clone()
```

Comment:
Ah, in that case we should add a `From<FixedInt> for Int` impl so you can do

```rust
self.input.consume(1).read_fixed_int(n)?.0.into()
```

and it will discard the `FixedInt` leaving the `Int` inside.